python-package/lightgbm/basic.py

```diff
@@ -3460,7 +3460,7 @@ def refit(self, data, label, decay_rate=0.9, **kwargs):
         ----------
         data : string, pathlib.Path, numpy array, pandas DataFrame, H2O DataTable's Frame or scipy.sparse
             Data source for refit.
-            If string or pathlib.Path, it represents the path to txt file.
+            If string or pathlib.Path, it represents the path to a text file (CSV, TSV, or LibSVM) or a LightGBM Dataset binary file.
```

Comment:
I don't think `refit()` supports binary data due that this data is used in `predict()` internally.

https://github.com/microsoft/LightGBM/blob/4e18c60b4d3bbcdb39f88c12f2001a1606cac268/python-package/lightgbm/basic.py#L3481

```py
import numpy as np
import lightgbm as lgb
from sklearn.datasets import load_boston

X, y = load_boston(return_X_y=True)
lgb_data = lgb.Dataset(X, y)
lgb_data.save_binary('dataset.bin')

bst = lgb.train({'objective': 'regression'}, lgb_data, num_boost_round=5)
bst.refit('dataset.bin', label=np.ones_like(y))
```

Reply:
ah you're right, bad copy-paste, sorry

Reply:
updated in https://github.com/microsoft/LightGBM/pull/4545/commits/33448b6d2a396447e6515b8357d69fa1cfc7c351, thanks for the sample code!